### ISSUE TITLE
Conceal additional commands like \section{} and \texttt{}

### ISSUE THREAD

**Issue by Melkster** [lervag/vimtex#1978]
VimTeX conceals some commands like for instance `\textbf{}`, `\textit{}` which I love.

Is there any plan to conceal more commands? The ones that I personally miss the most are for `\section{}`/`\subsection{}`/`\subsubsection{}` and `\texttt`.

For `\texttt{}` I think that the surrounding command should be concealed and the text inside it should get a syntax highlighting.

As for `\...section`, in order to differentiate between the different levels it could for instance get concealed into the markdown syntax like this:

**Without conceal:**

```tex
\section{First Section}
Foo bar

\subsection{Second Section}
Foo bar

\subsubsection{Third Section}
Foo bar
```

**With conceal:**

```markdown
# First Section
Foo bar

## Second Section
Foo bar

### Third Section
Foo bar
```

Would this be possible to implement?

**Comment by lervag** [comment]
> VimTeX conceals some commands like for instance `\textbf{}`, `\textit{}` which I love.

:)

(I personally don't like it, but I understand it is a popular feature!)

> Is there any plan to conceal more commands? The ones that I personally miss the most are for `\section{}`/`\subsection{}`/`\subsubsection{}` and `\texttt`.

No plan except that I'll be glad to improve as suggestions are provided, such as this.

> For `\texttt{}` I think that the surrounding command should be concealed and the text inside it should get a syntax highlighting.

Ok, so: `\texttt{asdf} -> asdf` where `asdf` is given a specific highlighting; which kind, if so? This command is already matched as `texCmdStyle`, so I would propose that the contents would be `texStyleArgTT`.

> As for `\...section`, in order to differentiate between the different levels it could for instance get concealed into the markdown syntax like this:
> ...
> Would this be possible to implement?

Actually, no: syntax conceals only allow a single substitution character of the concealed text. In this particular case, we can conceal the `\*section{...}`, but we can't use more than a single character. So, we could do something like this:

```tex
% before
\section{sec}
\subsection{subsec}

% after
¤sec
#subsec
```

The initial character can be chosen quite freely, but we can't add a space separator or use multiple characters.

Thus, I'm not sure if concealing the `\section` family of commands is a very good idea?

**Comment by lervag** [comment]
By the way, conceal of `\textttt` is also covered by #1968, so let's discuss that part there and let this discussion be about the `\section` stuff.

**Comment by Melkster** [comment]
> which kind, if so?

The highlight group is just a name right? In that case `texStyleArgTT` sounds good to me. I would propose that it by default was linked to `:hi String`. This is what [vim-markdown's](https://github.com/plasticboy/vim-markdown) highlight for inline code is linked to by default.

> syntax conceals only allow a single substitution character of the concealed text

After some tinkering I was able to throw together this proof of concept that turns `\section{Foo}` to `# Foo` by dividing the pattern into different parts. I based it on [this reddit thread](https://www.reddit.com/r/vim/comments/f8qwhu/concealing_with_more_than_one_character/fin3iit?utm_source=share&utm_medium=web2x&context=3) and [this stackoverflow answer](https://stackoverflow.com/a/55379732) :

```vim
syntax region Statement start='\\section{' end='}' transparent contains=section1,section2,section3
syntax match section1 '\\\%(section.*\)\@=' contained conceal cchar=#
syntax match section2 '\%(\\\)\@<=section{\ze\w\+' contained conceal cchar= " space
syntax match section3 '\%(section{\w\+\)\@<=}' contained conceal cchar= " space

set conceallevel=2
```

I'm not sure what the neatest way of getting it to also handle `\subsection` and `\subsubsection` is but hopefully this can help us achieve what we want.

**Comment by lervag** [comment]
This is now implemented in #2022; please test and give some feedback. I've also added a couple of questions that I'd like to hear opinions on.
